server/context.go

```diff
@@ -55,18 +55,29 @@ func (s *SessionManager) nextPid() uint64 {
 	return s.pid
 }
 
-// NewSession creates a Session for the given connection.
+// newSession creates a Session for the given connection.
+func (s *SessionManager) newSession(conn *mysql.Conn) sql.Session {
+	return s.builder(conn, s.addr)
+}
+
+// NewSession creates a Session for the given connection and saves it to
+// session pool.
 func (s *SessionManager) NewSession(conn *mysql.Conn) {
 	s.mu.Lock()
-	s.sessions[conn.ConnectionID] = s.builder(conn, s.addr)
+	s.sessions[conn.ConnectionID] = s.newSession(conn)
```

Comment:
i don't understand this change. Why do we need `s.builder` call in another function? It's just one line